code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -904,6 +904,29 @@ GLOBAL_LIST_EMPTY(PDAs)
 			old_id.forceMove(get_turf(src))
 
 
+/obj/item/pda/pre_attack(obj/target, mob/living/user, params)
+	if(ismachinery(target))
+		var/obj/machinery/target_machine = target
+		if(!target_machine.panel_open && !istype(target, /obj/machinery/computer))
+			return ..()
+		if(!istype(cartridge, /obj/item/cartridge/virus/clown))
+			return ..()
+		var/obj/item/cartridge/virus/installed_cartridge = cartridge
+
+		if(installed_cartridge.charges <=0)
+			to_chat(user, "<span class='notice'>Out of charges.</span>")
+			return ..()
```

Comment:
Say you want to use your PDA with an ID in it to open a door, like you can do normally. Will you always get this message then?

Reply:
Only if the door panel is open and you click on it with the PDA in your hand.